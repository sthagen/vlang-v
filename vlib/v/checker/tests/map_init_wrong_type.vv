fn main() {
   mut a := map[string]f32{}
   a = map{ 'x': 12.3 }
   _ = map{2:0 3:0 "hi":0}
   _ = map{2:0 3:`@` 4:0}
  _ = a
}
