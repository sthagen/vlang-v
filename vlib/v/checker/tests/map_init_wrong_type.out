vlib/v/checker/tests/map_init_wrong_type.vv:3:18: error: invalid map value: expected `f32`, not `float literal`
    1 | fn main() {
    2 |    mut a := map[string]f32{}
    3 |    a = map{ 'x': 12.3 }
      |                  ~~~~
    4 |    _ = map{2:0 3:0 "hi":0}
    5 |    _ = map{2:0 3:`@` 4:0}
vlib/v/checker/tests/map_init_wrong_type.vv:4:20: error: invalid map key: expected `int`, not `string`
    2 |    mut a := map[string]f32{}
    3 |    a = map{ 'x': 12.3 }
    4 |    _ = map{2:0 3:0 "hi":0}
      |                    ~~~~
    5 |    _ = map{2:0 3:`@` 4:0}
    6 |   _ = a
vlib/v/checker/tests/map_init_wrong_type.vv:5:18: error: invalid map value: expected `int`, not `rune`
    3 |    a = map{ 'x': 12.3 }
    4 |    _ = map{2:0 3:0 "hi":0}
    5 |    _ = map{2:0 3:`@` 4:0}
      |                  ~~~
    6 |   _ = a
    7 | }